cl/api/webhooks.py

```diff
@@ -202,3 +204,32 @@ def send_search_alert_webhook(
         content=post_content,
     )
     send_webhook_event(webhook_event, json_bytes)
+
+
+def send_pray_and_pay_webhooks(prayer: Prayer, webhook: Webhook) -> None:
+    """Send webhook event when a pray-and-pay request is granted.
+
+    :param prayer: The Prayer object representing the granted prayer.
+    :param webhook: The Webhook object to send the event to.
+    :return: None
+    """
+
+    # Only send webhook for granted prayers
+    if prayer.status != Prayer.GRANTED:
+        return
+
+    payload = PrayerSerializer(prayer).data
+    post_content = {
+        "webhook": generate_webhook_key_content(webhook),
+        "payload": payload,
+    }
+    renderer = JSONRenderer()
+    json_bytes = renderer.render(
+        post_content,
+        accepted_media_type="application/json;",
+    )
+    webhook_event = WebhookEvent.objects.create(
+        webhook=webhook,
+        content=post_content,
+    )
+    send_webhook_event(webhook_event, json_bytes)
```

Comment:
@albertisfu  suggested converting this method into a Celery task so we can send webhooks in the background and avoid blocking the post_save signal. To implement this, We need to tweak  several files, so it was difficult to list all the changes in a single comment.

I created a separate branch to encapsulate all the related updates. The following commits (in order) include the complete set of changes:

https://github.com/freelawproject/courtlistener/commit/60ad9ff061d423bb97bb018cf55f6f8a5eacf9c3

https://github.com/freelawproject/courtlistener/commit/ea78d17e2bcc23509d2a78fdbe3da0571c80afa6

https://github.com/freelawproject/courtlistener/commit/2994171102ba41606cf4ba283fe24970735520bf

Reply:
refactor is merged thanks @ERosendo